hnn_core/network_builder.py

```diff
@@ -376,30 +375,24 @@ def _create_cells_and_drives(self, threshold, record_vsoma=False,
         These drives are spike SOURCES but cells are also targets.
         External inputs are not targets.
         """
-        type2class = {'L2_pyramidal': pyramidal, 'L5_pyramidal': pyramidal,
-                      'L2_basket': basket, 'L5_basket': basket}
         # loop through ALL gids
         # have to loop over self._gid_list, since this is what we got
         # on this rank (MPI)
-
         for gid in self._gid_list:
             src_type, src_pos, is_cell = self.net._get_src_type_and_pos(gid)
 
-            if is_cell:  # not a drive
-                # figure out which cell type is assoc with the gid
-                # create cells based on loc property
+            if is_cell:
+                cell = Cell(name=_short_name(src_type), pos=src_pos, gid=gid)
+                p_secs = self.net.cell_properties[src_type]['sections']
+                p_syn = self.net.cell_properties[src_type]['synapses']
+                topology = self.net.cell_properties[src_type]['topology']
+                sect_loc = self.net.cell_properties[src_type]['sect_loc']
+                # instantiate NEURON object
+                cell.build(p_secs, p_syn, topology, sect_loc)
```

Comment:
Note that `cell.build()` is called here instead of e.g. in `cells_default.pyramidal()`.